%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%2345678901234567890123456789012345678901234567890123456789012345678901234567890
%        1         2         3         4         5         6         7         8

%\documentclass[letterpaper, 10 pt, conference]{ieeeconf}  % Comment this line out
                                                          % if you need a4paper
%
\documentclass[a4paper, 10pt, conference]{ieeeconf}      % Use this line for a4
                                                          % paper

\IEEEoverridecommandlockouts                              % This command is only
                                                          % needed if you want to
                                                          % use the \thanks command
\overrideIEEEmargins
% See the \addtolength command later in the file to balance the column lengths
% on the last page of the document



% The following packages can be found on http:\\www.ctan.org
\usepackage{graphics} % for pdf, bitmapped graphics files
%\usepackage{epsfig} % for postscript graphics files
%\usepackage{mathptmx} % assumes new font selection scheme installed
%\usepackage{times} % assumes new font selection scheme installed
%\usepackage{amsmath} % assumes amsmath package installed
%\usepackage{amssymb}  % assumes amsmath package installed
\newcommand{\horrule}[1]{\rule{\linewidth}{#1}} 	% Horizontal rule

%mypackages
\usepackage{amsmath}
 \usepackage[table,xcdraw]{xcolor}
\usepackage[pdftex]{graphicx}
\usepackage[english]{babel}
\usepackage{geometry} 
\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage{movie15}

\usepackage{fancyhdr} 
\usepackage{listings}
\definecolor{light-gray}{gray}{0.95}
\definecolor{pblue}{rgb}{0.13,0.13,1}
\definecolor{pgreen}{rgb}{0,0.5,0}
\definecolor{pred}{rgb}{0.9,0,0}
\definecolor{pgrey}{rgb}{0.46,0.45,0.48}
\lstset{language=Java,
		xleftmargin=0.5cm,
		framesep=4pt,
		framerule=0pt,
		stepnumber=1,
		numbersep=8pt,
		showstringspaces=false,
		breaklines=true,
		frameround=ftff,
		frame=single,
		belowcaptionskip=5em,
		belowskip=3em,
	showspaces=false,
	 backgroundcolor=\color{light-gray},
	showtabs=false,
	breaklines=true,
	showstringspaces=false,
	breakatwhitespace=true,
	commentstyle=\color{pgreen},
	keywordstyle=\color{pblue},
	stringstyle=\color{pred},
	basicstyle=\ttfamily
}


\usepackage{multicol}
\usepackage[hyperfootnotes=false]{hyperref}
\hypersetup{
	colorlinks,
	citecolor=black,
	filecolor=black,
	linkcolor=black,
	colorlinks=false,
	urlbordercolor=white,
	citebordercolor=black,
	linkbordercolor = red
}
\title{
		%\vspace{-1in} 	
		\usefont{OT1}{bch}{b}{n}
		\normalfont \normalsize \textsc{University of Illinois At Chicago\\CS559 - Neural Networks} \\ [25pt]
		\horrule{2pt} \\[0.4cm]
		\huge Project Outline \\
		\horrule{2pt} \\[0.3cm]
}
\author{
		\normalfont 								\large
        Umberto Di Fabrizio\\		\normalsize
        \today \\[0.5cm]
}
\date{}


\begin{document}

\maketitle
\thispagestyle{empty}
\pagestyle{empty}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\begin{abstract}
%The protein superfamily classification problem, which consists of determining the superfamily membership
%of a given unknown protein sequence, is very important for a biologist for many practical reasons, such as
%drug discovery, prediction of molecular function and medical diagnosis. In this work, we propose a new
%approach for protein classification based on a Probabilistic Neural Network and feature selection. Our goal
%is to predict the functional family of novel protein sequences based on the features extracted from the
%proteinâ€™s primary structure i.e., sequence only. For this purpose, the datasets are extracted form Protein
%Data Bank(PDB), a curated protein family database, are used as training datasets. In these conducted
%experiments, the performance of the classifier is compared to other known data mining approaches /
%sequence comparison methods. The computational results have shown that the proposed method performs
%better than the other ones and looks promising for problems with characteristics similar to the problem.
%\end{abstract}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{INTRODUCTION}
Bioinformatics has been growing in the last three decades\cite{Efficient} given the huge amount of biological data that is continuously gathered, mainly about DNA, RNA and proteins. The volume of data generated from project such as The Human Genome Project\cite{human}(1990-2003) has strengthen the collaboration between the computer scientist community and the biologist one.\\
One of the most challenging problem is to classify protein accordingly to their function or by their family or superfamily. Protein sequences are composed by an unique sequence of 20 amino acids which determines the protein function. They carry out fundamental roles for the cells functions ( basically represent the blueprint of the cell), infact they determine the shape and the structure of the cell.\\ Each protein encode a certain function which depends on its structure and amino acids sequence but can only be completely understood with experiments. Those experiments are costly and slow thus they cannot keep pace with the amount of information available and which needs to be annotated.\\
The challenge that has been tackled (TUTTE LE REFERENZE SU CLASSIFICAZIONE) is to classify proteins into functional or structural existing superfamilies so that the annotation process can be automated. \\  
A protein superfamily is a set of proteins for which common ancestry can be inferred so they possed sequence or structural homology. 
In a superfamily classification, an unlabeled protein sequence may belong
to any of the superfamily from a set of known superfamilies. The computational techniques analyze whether the protein belongs to any of the known superfamiles or whether it has no relation with any of them. This classification is enormously useful because similar protein sequences exhibit almost the same biological structure and function, more importantly one of the main reason is treating and preventing genetic disease as well as drug discovery, prediction of molecular function and medical diagnosis.

\section{METHODS}
Several methods have been investigated in order to solve the superfamily classification problem: determining the superfamily membership of a given unknown protein sequence.
BLAST (Basic Local Alignment Search Tool) [1] is a
tool that uses direct modelling, performing a search of homologie
between sequences. This software explores the local
alignment in pairs to measure the similarity between sequences.
The classification is done based on the alignment
which had the greatest punctuation.

Another method that uses direct modelling is the HMM
Hidden Markov Models that is widely used for probabilistic
modelling of family of proteins. It uses probabilistic values
to score how much an unkown protein belongs to a given
family.

A Fuzzy ARTMAP model, a machine learning method has been proposed used to classify the protein sequence\cite{fuzzy}.

The use of Neural Networks to tackle this problem has been successfully presented in the work by C.H.Wu at al.\cite{wu1992}\cite{wu1995}\cite{wu1996} and an introductory survey of neural networks applied to genome problems can be found in the book by C.H.Wu and Jerry W. McLarty\cite{bookWu}, the book explain the basic idea of neural networks presenting the different kinds of network that can be used and gives meaningful example on how to use them.\\
The process of protein classification using ANN can be
divided in three parts: (i) pre-processing: protein sequence
encoding; (ii) processing the protein classification using the
ANN. The first step is the most challenging, the reason is that protein sequences have different lengths and can even be very long (\textasciitilde
1000) whilst the neural network has a fixed amount of inputs and can hardly handle missing inputs. The methods of encoding can be basically dived in two types\cite{bookWu}: direct or indirect. The direct encoding basically translate each amino acid into a binary vector\footnote{other techniques will be presented later} accordingly to the one-hot encoding, this means that given that we have 20 possible amino acids then each of them will be represented by a vector of 20 bits with only one position at '1' and all the rest with '0'. Of course this kind of encoding is not feasible in the common case because a protein sequence with 300 amino acid will be translated with 20*300=6000 binary inputs.\\
The indirect encoding tries to extract useful features from the protein sequence in order to give meaningful inputs to the neural networks, one example is usually the n-gram hashing method\cite{wu1992}. This ngram method computes residue frequencies, the basic idead is that if we have a sequence \textit{Seq}='ACACTGAC' then the possible bi-gram are 'AC','AT','AG','CA,'CT','CG','TA,'TC','TG' and for each of them the occurrences are counted, usually the approach involves also the mapping of the original sequence to a smaller alphabet. A summary table on the encoding pros and cons is shown in Table I\\

\begin{table*}[t]
\centering
\label{enc}
\caption{Comparison between direct and indirect protein sequence encoding}
\begin{tabular}{|
		>{}c |c|c|}
	\hline
	& Pros             & Cons                                                                                                                                  \\ \hline
	Direct   & Keeps the sequence order                 & Encoding is too large                                                                                                                                         \\ \hline
	Indirect & Independent of length of the biosequence & \begin{tabular}[c]{@{}c@{}}It is hard to find meaningful feature,\\ this requires domain knowledge.\\ The order of the sequence is not mantained\end{tabular} \\ \hline
\end{tabular}



\end{table*}

\newpage

\section{PROPOSED SOLUTION}
The idea is to exploit the power of the Deep Learning Neural Network in order to classify the protein in its family.
The work will follow the following outline:

\begin{itemize}
	\item \textbf{Data collection}\\
	We want to obtain for each superfamily or family the list of all the proteins that belong to that family and their complete protein sequence.\\
	This step is particularly tricky because there is no a database of protein sequences and families, or better there are several but they all have to be quired manually through a web interface. During this phase the data will be scraped from the web sites (\href{http://prosite.expasy.org/}{http://prosite.expasy.org/}, \href{http://www.uniprot.org/uniprot/}{http://www.uniprot.org/uniprot/}) with the use of the Kimono platform\cite{kimono} and Rscripts. Automatizing this process will come in handy later, or if any new superfamily data has to be collected.
	\item \textbf{Data Encoding}\\
	During this phase the sequences of proteines will be analyzed and encoded through the indirect method. Different solution will be explored as regard the n-gram value (2,3) and the possibility to include any other valuable information. The objective is to extract as many information as possible from the protein without creating too many inputs for the neural network.
	\item \textbf{Design of the NN}
	The input of the neural network will be the encoded protein sequence (the estimated size will be around 500 inputs), the output is the superfamily of the protein or 'Other' if the protein does not belong to any of the available families (possibly 3 or 4).
	The neural network two design are still uncertain, the basic approach will be to first create a backpropagation network in order to have a classifier to use as baseline and to test the underlining assumptions and the encoding method.\\
	Then a Convolutional Neural Network will be designed, during the step the input format will be re-formatted from the vector form to the matrix form that CNN handles and the choice of the convolution filters and pooling method will be made by comparing different solution.\\
	At this point, if I will still have time the LAMSTAR will be designed and used although given the amount of time available this step may be omitted.
\end{itemize}

\clearpage
\newpage
\begin{thebibliography}{99}

\bibitem{graupe}D.Graupe, Principles of Artificial Neural Networks, 3nd ed., Advanced Series in Circuits and System-Vol.7

\bibitem{Efficient}Efficient Feature Selection and Classification of Protein Sequence Data in Bioinformatics,Muhammad Javed Iqbal et al., 2014

\bibitem{human}\href{https://www.genome.gov/12011238}{https://www.genome.gov/12011238}

\bibitem{blast}Basic local alignment tool,S. Altschul et al., 1990
\bibitem{wu1992}Protein classification artificial neural system, C. H. Wu et al, 1992
\bibitem{wu1995}Neural Networks for Full-Scale Protein Sequence, C. H. Wu et al, 1995
\bibitem{wu1996}Motif identification neural design for rapid and sensitive protein family search, C. H. Wu et al,1996
\bibitem{fuzzy}Multi-class Protein Sequence Classification Using Fuzzy ARTMAP,Shakir Mohamed et al.,2006

\bibitem{bookWu}Neural Networks and Genome Informatics
\bibitem{kimono}\href{https://www.kimonolabs.com/}{https://www.kimonolabs.com/}
\end{thebibliography}

\end{document}
